herepy/routing_api.py

```diff
@@ -178,3 +178,89 @@ def truck_route(self,
         if modes is None:
             modes = [RouteMode.truck, RouteMode.fastest]
         return self._route(waypoint_a, waypoint_b, modes)
+
+
+
+class InvalidCredentialsError(HEREError):
+
+    """Invalid Credentials Error Type.
+
+    This error is returned if the specified token was invalid or no contract
+    could be found for this token.
+
+    """
+
+
+class InvalidInputDataError(HEREError):
```

Comment:
If we can use new error classes in general I think it's better to locate them in error.py.

Reply:
I think the APIs are developed by completely separate teams and they are behaving extremely different in terms of communication patterns, documentation and so on.

The Routing API is the only API which offers Error Types as far as I could find in the documentation: https://developer.here.com/documentation/routing/topics/resource-types-error.html

The Places API for example does not use schemas etc but returns
```json
{
    "status": 401,
    "message": "Invalid app_id app_code combination"
}
````

As the APIs are clearly disconnected I would also separate the Error Classes.

Reply:
Ok, no worries.